service/history/api/updateworkflow/api.go

```diff
@@ -65,9 +66,10 @@ type Updater struct {
 	req                        *historyservice.UpdateWorkflowExecutionRequest
 	namespaceID                namespace.ID
 
-	wfKey     definition.WorkflowKey
-	upd       *update.Update
-	directive *taskqueuespb.TaskVersionDirective
+	wfKey                  definition.WorkflowKey
+	upd                    *update.Update
+	directive              *taskqueuespb.TaskVersionDirective
+	workflowVersioningInfo *deploymentpb.WorkflowVersioningInfo
```

Comment:
can you move both of this under line 80. They need to be under this WARNING comment below.

Reply:
workflowVersioningInfo is removed and u.directive does not come from MS, it's created fresh.